gpkit/nomials/nomial_math.py

```diff
@@ -715,8 +718,10 @@ def __init__(self, left, right):
     def as_posyslt1(self):
         "Returns the posys <= 1 representation of this constraint."
         # todo deal with substitutions
-        raise TypeError("SignomialEquality could not simplify to"
-                        " a PosynomialInequality")
+        raise InvalidGPConstraint("SignomialEquality could not simplify"
+                                  " to a PosynomialInequality; try forming"
+                                  " your program as SignomialProgram or"
+                                  " calling .localsolve().")
```

Comment:
When we're forming an SP (or trying and failing to form a GP from an SP) we don't care about SignomialsEnabled.